writings/tests/test_models.py

```diff
@@ -1,38 +1,127 @@
+import datetime
+
 from django.test import TestCase
 from django.db.utils import IntegrityError
-from .factories import (
-    CategoryFactory,
-    EntryFactory,
-    EntryCollectionFactory,
-    OrderedEntryFactory,
-    CollectionWithSortedEntriesFactory,
-)
+from django.utils.timezone import now
+
+from . import factories
 
 
 class EntryTestCase(TestCase):
 
     def test_entry_auto_publication_date(self):
-        obj = EntryFactory.create(published=True)
+        """Publication date should be set on save."""
 
+        obj = factories.EntryFactory(published=True, publication_date=None)
         self.assertIsNotNone(obj.publication_date)
 
     def test_entry_categories(self):
-        CategoryFactory.create_batch(4)
-        obj = EntryFactory.create(categories=CategoryFactory.create_batch(2))
+        factories.CategoryFactory.create_batch(4)
+        obj = factories.EntryFactory.create(categories=factories.CategoryFactory.create_batch(2))
 
         self.assertEqual(obj.categories.count(), 2)
 
+    def test_url(self):
+        """Entries should know their own URL."""
+
+        entry = factories.EntryFactory(title='My Awesome Entry')
+        self.assertEqual(entry.get_absolute_url(), '/analysis/entries/my-awesome-entry/')
```

Comment:
We can but I sometimes feel weird about it. Feels like a tautology. Happy to update these since that's more consistent with the existing tests.